Added all lexer symbols. Added integer declaration parsing

lotrbuilders opened the pull request:
Unfortunately couldn't automatically merge. Should give error message when parsing any statement as the evaluation expects C89 compatibility at the moment

lotrbuilders committed:
lexer+ parser expansion

lexer extended to except keywords.
Parser extended to accept integer decleration

lotrbuilders committed:
Added all symbols to lexer. Local integer declarations should be evaluatable

only local integers are newly parsed.  Intentionally broke compatibility with statement and declaration list as they were for C89 compatibility

chorman0773 commented on src/lexer.l:
```diff
@@ -14,12 +14,54 @@ D   [0-9]
+"static"	{return STATIC;}
+"auto"		{return AUTO;}
+"volatile"	{return VOLATILE;}
+"_Atomic"	{return _ATOMIC;}
```
_ATOMIC is reserved for the implementation. I recommend using ATOMIC instead

lotrbuilders committed:
Merge branch 'master' of git://github.com/WorldsApartDevTeam/65816-c

lotrbuilders committed:
Expanded parser to all statements

All statements and all expresions that don't use type-identifier are parseable. Declarations have not changed. Strings are also identified

lotrbuilders commented:
I removed the objects and changed _ATOMIC to ATOMIC. Also added new features
